browser/src/layer/marker/TextInput.js

```diff
@@ -1014,6 +1041,129 @@ L.TextInput = L.Layer.extend({
 			this._textArea.setAttribute('data-accept-input', accept);
 		}
 		this._acceptInput = accept;
+	},
+
+	_hasPreSpace: function() {
+		var preSpace = document.getElementById('pre-space');
```

Comment:
Hmm - can we avoid searching the whole document, and just search in this._textArea; I'm not a huge fan of this "lets search the huge global variable for a string called 'fish'" or whatever =)

Reply:
No problem I will rewrite it

Reply:
Quite possibly we could use a local member variable this.preSpace - which holds a pointer to the element and managing that nicely (?)

Reply:
New rewrite just check textArea first/last child